org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/FavoriteSimpleTreeModel.java

```diff
@@ -0,0 +1,460 @@
+				ZkCssHelper.appendStyle(tc, "color: #" + hex);
+			}
+
+			// Set Icon
+			if (ThemeManager.isUseFontIconForImage())
+				tc.setIconSclass(getIconSclass(mNode));
+			else
+				tc.setImage(ThemeManager.getThemeResource(getIconFile(mNode)));
+
+			//
+			if (mNode.isSummary())
+			{
+				tc.addSclass("fav-summary-folder");
+			}
+
+			//
+			if (mNode.isWindow())
+			{
+				// Check Window access for ReadWrite & New Toolbar button
+				if (!MToolBarButtonRestrict.isNewButtonRestricted(MMenu.get(mNode.getMenu_ID()).getAD_Window_ID()))
+				{
+					Toolbarbutton newBtn = new Toolbarbutton();
+					newBtn.setTooltiptext(Util.cleanAmp(Msg.getMsg(Env.getCtx(), "New")));
+					newBtn.addEventListener(Events.ON_CLICK, this);
+					tc.appendChild(newBtn);
+
+					if (ThemeManager.isUseFontIconForImage())
+					{
+						newBtn.setIconSclass("z-icon-New");
+						newBtn.setSclass("new-toolbarbutton");
+					}
+					else
+					{
+						newBtn.setSclass("menu-href-newbtn");
+						newBtn.setImage(ThemeManager.getThemeResource("images/New10.png"));
+					}
+				}
+			}
+
+			if (ClientInfo.isMobile())
+			{
+				Toolbarbutton btnContextMenu = new Toolbarbutton();
+				btnContextMenu.setClass("fav-mobile-ctx-menu");
+				btnContextMenu.setIconSclass("z-icon-More");
+				btnContextMenu.setAttribute(MOBILE_TOOLBAR_CTX_MENU, true);
+				btnContextMenu.addEventListener(Events.ON_CLICK, this);
+				tc.appendChild(btnContextMenu);
+			}
+		}
+		else
+		{
+			tr = ti.getTreerow();
+			tr.getChildren().clear();
+		}
+		tc.setParent(tr);
+		ti.setValue(node);
+	} // render
+
+	@Override
+	public void onEvent(Event event) throws Exception
+	{
+		Component comp = event.getTarget();
+		String eventName = event.getName();
+
+		if (Events.ON_DROP.equals(eventName)	|| Events.ON_RIGHT_CLICK.equals(eventName)
+			|| (Events.ON_CLICK.equals(eventName)	&& comp instanceof Toolbarbutton
+				&& (boolean) ((Toolbarbutton) event.getTarget()).getAttribute(FavoriteSimpleTreeModel.MOBILE_TOOLBAR_CTX_MENU)))
+		{
+			for (EventListener<Event> listener : onDropListners)
+			{
+				listener.onEvent(event);
+			}
+		}
+		// On click of menu to open that window
+		else if (Events.ON_CLICK.equals(eventName) || Events.ON_SELECT.equals(eventName))
+		{
+			boolean newRecord = false;
+			if (comp instanceof Toolbarbutton)
+			{
+				comp = comp.getParent().getParent();
+				newRecord = true;
+			}
+
+			if (comp instanceof Treerow)
+			{
+				Treerow treerow = (Treerow) comp;
+				Treeitem treeitem = (Treeitem) treerow.getParent();
+				Object value = treeitem.getValue();
+
+				DefaultTreeNode<?> dtNode = (DefaultTreeNode<?>) value;
+				MTreeNode mtn = (MTreeNode) dtNode.getData();
+
+				if (newRecord)
+				{
+					onNewRecord(mtn.getMenu_ID());
+				}
+				else if (!mtn.isSummary())
+				{
+					int menuID = mtn.getMenu_ID();
+					SessionManager.getAppDesktop().onMenuSelected(menuID);
+				}
+			}
+		}
+		else if (Events.ON_DOUBLE_CLICK.equals(eventName))
```

Comment:
This event doesn't seem necessary because there is an Edit Folder dedicated button.

Double clicking in folder is throwing console error:
Msg.getMsg: NOT found: EditFolderName

I think better to remove this event.

If this is removed then the changes on WTextEditorDialog for IsShowHTMLTab would be unnecessary.

However they look OK to integrate as it can be an improvement for other programs.
@hengsin - are you OK with that?  Or better to do that in its own ticket?

Reply:
@hengsin Can you please share your thought? I suggest we keep isShowHTMLTab in this commit if it is acceptable otherwise, we have to make two different PR for User favorite, one without this flag and one after we add this support on WTextEditor.

Reply:
it is a small change, I'm fine with that.